src/main/scala/com/target/data_validator/validator/ColumnBased.scala

```diff
@@ -62,41 +83,61 @@ case class ColumnMaxCheck(column: String, value: Json)
 
   override def configCheck(df: DataFrame): Boolean = checkTypes(df, column, value)
 
+  // scalastyle:off
   override def quickCheck(row: Row, count: Long, idx: Int): Boolean = {
     val dataType = row.schema(idx).dataType
     val rMax = row(idx)
     logger.info(s"rMax: $rMax colType: $dataType value: $value valueClass: ${value.getClass.getCanonicalName}")
-    val num = value.asNumber
-    failed = dataType match {
-      case StringType => value.asString.exists(_ != row.getString(idx))
-      case ByteType => num.map(_.toByte).exists(_.get != row.getByte(idx))
-      case ShortType => num.map(_.toShort).exists(_.get != row.getShort(idx))
-      case IntegerType =>
-        val intNum = value.asNumber.map(_.toInt.get).getOrElse(-1)
-        val rowInt = row.getInt(idx)
-        logger.debug(s"intNum[${intNum.getClass.getCanonicalName}]: $intNum " +
-          s"rowInt[${rowInt.getClass.getCanonicalName}]: $rowInt")
-        num.map(_.toInt).exists(_.get != row.getInt(idx))
-      case LongType => num.map(_.toLong).exists(_.get != row.getLong(idx))
-      case FloatType => num.forall(_.toDouble != row.getFloat(idx))
-      case DoubleType => num.forall(_.toDouble != row.getDouble(idx))
-      case ut =>
-        logger.error(s"quickCheck for type: $ut, Row: $row not Implemented! Please file this as a bug.")
-        true // Fail check!
+
+    var errorMsg = ""
+    val data = LinkedHashMap.empty[String, String]
+
+    def resultForString(): Unit = {
+      val (expected, actual) = (value.asString.getOrElse(""), row.getString(idx))
+
+      failed = expected != actual
+      data += ("expected" -> expected, "actual" -> actual)
+      errorMsg = s"ColumnMaxCheck $column[StringType]: Expected: $expected, Actual: $actual"
     }
+
+    def resultForNumeric(): Unit = {
+      val num = value.asNumber.get
+      var cmp_params = (0.0, 0.0) // (expected, actual)
+
+      dataType match {
+        case ByteType => cmp_params = (num.toByte.getOrElse[Byte](-1), row.getByte(idx))
+        case ShortType => cmp_params = (num.toShort.getOrElse[Short](-1), row.getShort(idx))
+        case IntegerType => cmp_params = (num.toInt.getOrElse[Int](-1), row.getInt(idx))
+        case LongType => cmp_params = (num.toLong.getOrElse[Long](-1), row.getLong(idx))
+        case FloatType => cmp_params = (num.toDouble, row.getFloat(idx))
+        case DoubleType => cmp_params = (num.toDouble, row.getDouble(idx))
+      }
+
+      failed = cmp_params._1 != cmp_params._2
+      val pctError = if(failed) calculatePctError(cmp_params._1, cmp_params._2) else "0.00%"
+      data += ("expected" -> num.toString, "actual" -> rMax.toString, "error_percent" -> pctError)
+      errorMsg = s"ColumnMaxCheck $column[$dataType]: Expected: $num, Actual: $rMax. Error %: ${pctError}"
```

Comment:
Again, the redundant percentage symbol and let's be specific that this is relative error.  Also, this is basically the duplicate format for `MinNumRowsCheck` with some slight differences (the period and comma here).  This can be defined once and then used both cases.

```suggestion
      errorMsg = s"ColumnMaxCheck $column[$dataType]: Expected: $num, Actual: $rMax Relative Error: ${pctError}"
```